server/src/main/java/com/objectcomputing/checkins/services/checkins/CheckInServicesImpl.java

```diff
@@ -18,62 +20,80 @@
-            if(checkIn.getId()!=null){
-            throw new CheckInBadArgException(String.format("Found unexpected id for checkin  %s", checkIn.getId()));
-        } else if(memberId.equals(pdlId)){
-            throw new CheckInBadArgException(String.format("Team member id %s can't be same as PDL id", checkIn.getTeamMemberId()));
-        } else if(!memberRepo.findById(memberId).isPresent()){
-            throw new CheckInBadArgException(String.format("Member %s doesn't exists", memberId));
-        } else if(chkInDate.isBefore(LocalDate.EPOCH) || chkInDate.isAfter(LocalDate.MAX)) {
-            throw new CheckInBadArgException(String.format("Invalid date for checkin %s",memberId));
-        }
-        checkInRet = checkinRepo.save(checkIn);
+
+            if(checkIn.getId()!=null) {
+                throw new CheckInBadArgException(String.format("Found unexpected id for checkin  %s", checkIn.getId()));
+            } else if(memberId.equals(pdlId)) {
+                throw new CheckInBadArgException(String.format("Team member id %s can't be same as PDL id", checkIn.getTeamMemberId()));
+            } else if(!memberRepo.findById(memberId).isPresent()) {
+                throw new CheckInBadArgException(String.format("Member %s doesn't exists", memberId));
+            } else if(!pdlId.equals(memberRepo.findById(memberId).get().getPdlId())) {
+                throw new CheckInBadArgException(String.format("PDL %s is not associated with member %s", pdlId, memberId));
+            } else if(chkInDate.isBefore(LocalDate.EPOCH) || chkInDate.isAfter(LocalDate.MAX)) {
+                throw new CheckInBadArgException(String.format("Invalid date for checkin %s",memberId));
+            }
+            checkInRet = checkinRepo.save(checkIn);
         }
-            return checkInRet ;
+
+        return checkInRet ;
     }
 
 
     @Override
     public CheckIn read(@NotNull UUID id) {
-
         return checkinRepo.findById(id).orElse(null);
     }
 
     @Override
     public CheckIn update(CheckIn checkIn) {
+
         CheckIn checkInRet = null;
-        if(checkIn!=null){
-        final UUID id = checkIn.getId();
-        final UUID memberId = checkIn.getTeamMemberId();
-        LocalDate chkInDate = checkIn.getCheckInDate();
-        if(id==null||!checkinRepo.findById(id).isPresent()){
-            throw new CheckInBadArgException(String.format("Unable to find checkin record with id %s", checkIn.getId()));
-        }else if(!memberRepo.findById(memberId).isPresent()){
-            throw new CheckInBadArgException(String.format("Member %s doesn't exist", memberId));
-        } else if(memberId==null) {
-            throw new CheckInBadArgException(String.format("Invalid checkin %s", checkIn));
-        } else if(chkInDate.isBefore(LocalDate.EPOCH) || chkInDate.isAfter(LocalDate.MAX)) {
-            throw new CheckInBadArgException(String.format("Invalid date for checkin %s",memberId));
+
+        if(checkIn!=null) {
```

Comment:
As with the save method would it make sense to simply not allow this to be null?

Reply:
I'll make the required changes.